algorithms/kernel/objective_function/mse/mse_dense_default_batch_impl.i

```diff
@@ -62,64 +63,7 @@ inline services::Status MSEKernel<algorithmFPType, method, cpu>::compute(Numeric
         {
             if (xNT != dataNT)
             {
-                xNT                                               = dataNT;
-                SOANumericTable * soaDataPtr                      = dynamic_cast<SOANumericTable *>(dataNT);
-                HomogenNumericTable<algorithmFPType> * hmgDataPtr = dynamic_cast<HomogenNumericTable<algorithmFPType> *>(dataNT);
-                if (hmgDataPtr)
-                {
-                    X              = hmgDataPtr->getArray();
-                    transposedData = false;
-                }
-                else if (soaDataPtr)
-                {
-                    const bool featuresEqual                            = soaDataPtr->getDictionary()->getFeaturesEqual();
-                    bool isContinuous                                   = false;
-                    data_management::features::IndexNumType currentType = features::internal::getIndexNumType<algorithmFPType>();
-                    if (currentType == (*(soaDataPtr->getDictionary()))[0].getIndexType())
-                    {
-                        isContinuous = true;
```

Comment:
Did you check performance on continuous columnar based input data (SOA table with continuous data)?